docs/docs/data-fetching.md

```diff
@@ -0,0 +1,169 @@
+
+This combination of static/dynamic is possible through React [hydration](/docs/glossary#hydration), which means that Gatsby (through React.js) builds static files to render server-side. When Gatsby's script bundle downloads and executes in the browser, it preserves the HTML markup built by Gatsby and turns the site into a full React web application that can manipulate the [DOM](/docs/glossary#dom). The result of this process creates fast loading pages and a nice user experience.
+
+> To understand how statically generated content can turn into a React app, refer to the [Understanding React Hydration guide](/docs/react-hydration)
+
+Compiling pages at build time is useful when your website content won't change often, or when triggering a build process to recompile works fine. However, some websites with more dynamic needs require a [client-side](/docs/glossary#client-side) runtime to handle constantly changing content after the page loads, like a chat widget, user upvotes, or an email client web application.
+
+## Combining build time and client runtime data
+
+To illustrate this combination of build time and runtime data, this guide uses code from a [small example site](https://gatsby-data-fetching.netlify.com). It uses the [`gatsby-source-graphql`](/packages/gatsby-source-graphql/) plugin to fetch data from the GitHub API at build time for static content like the name and url to a repository, and the [`fetch` API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to retrieve more dynamic data from the GitHub API on the [client-side](/docs/glossary#client-side) like star counts when the page loads in the browser.
+
+> Check out the code from the [full example here](https://github.com/gatsbyjs/gatsby/tree/master/examples/data-fetching).
+
+### Fetching data at build time
+
+For fetching data at build time, you can create an integration with a third-party system by sourcing data in your `gatsby-node` file and creating nodes for the GraphQL layer that become queryable in pages. This is the same method that source plugins implement to [source data](/docs/content-and-data/) while the site builds. You can read about that process in the [Creating a Source Plugin guide](/docs/creating-a-source-plugin/).
+
+> This process of fetching data at build time and creating pages from the data is [covered in more depth in the tutorial](/tutorial/part-five/) as well as the docs for [creatiing pages from data programmatically](/docs/programmatically-create-pages-from-data/).
+
+#### Source data to be queried at build time
+
+The simplest way to source data is to take advantage of an existing source plugin. To install the `gatsby-source-graphql`:
```

Comment:
This is only simple if a) the user understands the process and b) if the API they want to hit has a GraphQL server. The latter seems like a caveat we should mention, and provide suggestions of alternatives if they want to retrieve data from somewhere at build time that isn't a GraphQL API endpoint.

Reply:
Very good point! Definitely an assumption, I'll reword this and provide other options.